packages/astro/src/runtime/server/render/component.ts

```diff
@@ -261,7 +261,7 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 
 			if (isPage || renderer?.name === 'astro:jsx') {
 				yield html;
-			} else {
+			} else if (html) {
```

Comment:
Could this cause the server to hang if there is _not_ HTML? Could we have an `else` case with a detailed internal error, or does fall-through work?

Reply:
yeah you are right i replace it with an empty string return now, that should do.